tests/lib/state_test.py

```diff
@@ -240,6 +240,39 @@ def test_absent_cannot_remove_absent(self):
         obj2 = state.RouteEntry(routes[0])
         assert not obj1.is_match(obj2)
 
+    def test_state_iface_routes(self):
+        routes = _get_mixed_test_routes()
+        route_state = state.State(
+            {
+                Route.KEY: {
+                    Route.CONFIG: routes
+                }
+            }
+        )
+
+        for route in routes:
+            assert (route in
+                    route_state.config_iface_routes.get(
+                        route[Route.NEXT_HOP_INTERFACE]))
+
+    def test_state_iface_routes_order(self):
+        # Changing all routes to eth1
+        routes = _get_mixed_test_routes()
+        for route in routes:
+            route[Route.NEXT_HOP_INTERFACE] = 'eth1'
+
+        route_state = state.State(
+            {
+                Route.KEY: {
+                    Route.CONFIG: sorted(routes, key=_route_sort_key,
+                                         reverse=True)
+                }
+            }
+        )
+
+        assert (sorted(routes, key=_route_sort_key) ==
```

Comment:
How about creating two route states, one sorted regularly and the other reversely and then compare the indexing result?
It would have been enough to just do a simpler one:
```
{Route.CONFIG: routes: [routes[0], routes[1]]}
{Route.CONFIG: routes: [routes[1], routes[0]]}
```
This will also remove the need for the logic duplication in the tests (`_route_sort_key`).